api/types/sortby.go

```diff
@@ -0,0 +1,22 @@
+package types
+
+import (
+	"strings"
+)
+
+// GetSortByFromString expects a string in format `<fieldName>:<asc|desc>` where
+// index 0 is fieldName and index 1 is direction.
+// If a direction is not set, or is not recognized, it defaults to ASC.
+func GetSortByFromString(sortStr string) (sortBy SortBy) {
+	if sortStr != "" {
```

Comment:
I would invert this check, so that most of the code is left-aligned:

```
if sortStr == "" {
    return sortBy
}
vals := strings.Split(...
...
```